### PR TITLE
Detail view

### LINKED ISSUE #11: [Story] Project detail view
# Value statement

As a user
I need to see additional information about a project
so that I can find out more about it

# Description (Text, Scribble, Wireframe, Design)

image

-  Upon click on a project, a card scoots up from the bottom and overlays the screen
-  It shows additional content such as affiliated people, company information, comments and updates

# Acceptance criteria

- [x] Detail page overlays blurred background
- [x] Press on back removes page
- [x] Information sections are expandable
- [ ] People section overlaps when over a certain number
- [ ] Page rises from the lower edge of the screen
Animations will be handled in the next story

# Tasks

- [x] TESTS 
- [x] Storybook
- [x] Create Overlay component -> receives all necessary project data
- [x] toggle on project click
- [x] people section
- [x] information section

# Complexity
medium

### PR DESCRIPTION
**Pull request by constantingoeldel**
This pull request resolves story #11

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,24 +5,35 @@ import TagCluster from './components/TagCluster'
...
   return (
     <>
+      {detailedProject && <Overlay project={detailedProject} onBack={toggleDetailOverlay} />}
+
       <Search onSearch={onSearch} />
       <Counter firstInt={projects.length} secondInt={mockData.projects.length} />
       <TagCluster tags={mockData.tags} onTagClick={onTagClick} />
       {projects.map((project, index) => (
-        <Project key={index} data={project} />
+        <Project key={index} data={project} onClick={toggleDetailOverlay} />
```
Comment (mpagels) [inline src/App.js]: Use of index as key is not a good practice. Do you have an other unique property? Is in the project obj a unique property available?

File: src/components/Contributors.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  )
+}
+Contributors.propTypes = {
+  contributors: PropTypes.arrayOf(
+    PropTypes.shape({
+      name: PropTypes.string,
+      picture: PropTypes.string,
+    })
+  ),
+}
```
Comment (mpagels) [inline src/components/Contributors.js:26]: maybe consider to put the PropTypes direct after the imports, so other devs can easily see, what types you are using. it's personal preference. I would like that

File: src/components/Overlay.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        {project.tags.map((tag, index) => (
+          <Tags content={tag} key={index} />
+        ))}
+      </TagList>
+
+      <Contributors contributors={project.contributors} />
+      <Detail title='Mission' content={project.mission} />
+      <Detail title='About' content={project.about} />
+      <Detail title='Updates' content={project.updates} />
+      <Detail title='Comments' content={project.comments} />
```
Comment (mpagels) [inline src/components/Overlay.js]: If the content property in project is only important for Detail component, you could put these in an array in your project object and map over it to display it.

File: src/components/Search.js [hunk 4]
```diff
@@ -37,7 +37,9 @@ const Input = styled.input`
   box-sizing: border-box;
   width: 70%;
 `
-const Button = styled.button`
+
+const Button = styled.button.attrs((props) => ({ type: 'button' }))`
```
Comment (mpagels) [inline src/components/Search.js:39]: nice ;-)

**Review by mpagels — COMMENTED** [review]
Very good job.

Short overview:
don't use index as key, try to find some unique value or use uuid
personal suggestion: put PropTypes after import, so devs who read your code for the first time, can see at first, what types you're using

found some tiny things.

pls review

File: src/components/Overlay.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+export default function Overlay({ project, onBack }) {
+  return (
+    <OverlaySection>
+      <Headline>
+        <ArrowStyled onClick={onBack} />
+        <Title>{project.title}</Title>
+        <ShareStyled />
+      </Headline>
+      <Img alt='' src={project.image ?? 'https://source.unsplash.com/random'}></Img>
```
Comment (LebkuchenCJ) [inline src/components/Overlay.js:18]: Alt attribtue missing

File: src/components/Overlay.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      <TagList>
+        {project.tags.map((tag, index) => (
+          <Tags content={tag} key={tag} />
```
Comment (LebkuchenCJ) [inline src/components/Overlay.js:21]: Tag doesn`t sound very unique to me. Like Martin suggested you go for ID as a key value. (If available)

File: src/components/Overlay.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  z-index: 99;
+  background-color: white;
+  border-radius: 10px 10px 0 0;
+  box-shadow: 0 -1px 10px rgba(46, 46, 58, 0.5);
+  padding: 10px;
+`
+const Img = styled.img`
+  margin-top: 50px;
+  width: 100%;
+  height: 250px;
```
Comment (LebkuchenCJ) [inline src/components/Overlay.js:68]: Fixed pixel values can be tricky on diffrent screen sizes. Maybe you can go for % as in your height?

**Review by LebkuchenCJ — APPROVED** [review]
Hey, great progress so far :) GJ
Some small things in general while opening you app.
I have following error in my console: 
`Error while trying to use the following icon from the Manifest: https://unfinished-kgfymk36f.vercel.app/logo192.png (Download error or resource isn't a valid image)`

While opening your app on my desktop I find it tricky to navigate as there no hover effect with my cursor on any button. Also, the tag elements didn`t look like they were clickable. 

The tags on your card elements are scrollable to the right. Have you tried making them smaller and make them all viewable at once?

Following items are not aligned to each other. Is this on purpose for any upcoming styling?
image
 
There is a little bit of space left at the top. if you want to have a floating effect you could add some space at the bottom, too. Or remove the space.
image

if any of the above comments marking things that are too early to mark, please ignore my Geschwafel :) 

Keep up

File: src/components/Overlay.js (hunk 3 above)
Comment (constantingoeldel) [reply src/components/Overlay.js]: Done

File: src/components/Overlay.js (hunk 6 above)
Comment (constantingoeldel) [reply src/components/Overlay.js:21]: I will use IDs when implement the backend, this needs to be sufficient until then

File: src/components/Overlay.js (hunk 5 above)
Comment (constantingoeldel) [reply src/components/Overlay.js:18]: I don't have any way to know what these pictures represent so I can't set an alt attribute

File: src/components/Overlay.js (hunk 7 above)
Comment (constantingoeldel) [reply src/components/Overlay.js:68]: I will have a general design improvement story sometime soon where I will address these issues and the aforementioned ones. Thanks for your code review!

**Pull request merged**